python/simulators/PyBlackOilSimulator.cpp

```diff
@@ -0,0 +1,162 @@
+/*
+  Copyright 2020 Equinor ASA.
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "config.h"
+#include <opm/parser/eclipse/Deck/Deck.hpp>
+#include <opm/parser/eclipse/EclipseState/EclipseState.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/Schedule.hpp>
+#include <opm/parser/eclipse/EclipseState/SummaryConfig/SummaryConfig.hpp>
+#define FLOW_BLACKOIL_ONLY
+#include <opm/simulators/flow/Main.hpp>
+#include <opm/simulators/flow/FlowMainEbos.hpp>
+// NOTE: EXIT_SUCCESS, EXIT_FAILURE is defined in cstdlib
+#include <cstdlib>
+#include <iostream>
+#include <string>
+#include <opm/simulators/flow/python/PyBlackOilSimulator.hpp>
+
+namespace py = pybind11;
+
+namespace Opm::Pybind {
+PyBlackOilSimulator::PyBlackOilSimulator( const std::string &deckFilename)
+    : deckFilename_{deckFilename}
+{
+}
+
+PyBlackOilSimulator::PyBlackOilSimulator(
+    std::shared_ptr<Opm::Deck> deck,
+    std::shared_ptr<Opm::EclipseState> state,
+    std::shared_ptr<Opm::Schedule> schedule,
+    std::shared_ptr<Opm::SummaryConfig> summary_config
+)
+    : deck_{std::move(deck)}
+    , eclipse_state_{std::move(state)}
+    , schedule_{std::move(schedule)}
+    , summary_config_{std::move(summary_config)}
+{
+}
+
+const Opm::FlowMainEbos<typename Opm::Pybind::PyBlackOilSimulator::TypeTag>&
+         PyBlackOilSimulator::getFlowMainEbos() const
+{
+    if (this->mainEbos_) {
+        return *(this->mainEbos_.get());
```

Comment:
Style: `std::shared_ptr<>` defines a dereference operation so you could just write this as
```C++
return *this->mainEbos_;
```
instead.